src/scene/graph-node.js

```diff
@@ -6,21 +6,21 @@ import { Mat4 } from '../math/mat4.js';
 import { Quat } from '../math/quat.js';
 import { Vec3 } from '../math/vec3.js';
 
-var scaleCompensatePosTransform = new Mat4();
-var scaleCompensatePos = new Vec3();
-var scaleCompensateRot = new Quat();
-var scaleCompensateRot2 = new Quat();
-var scaleCompensateScale = new Vec3();
-var scaleCompensateScaleForParent = new Vec3();
-var tmpMat4 = new Mat4();
-var tmpQuat = new Quat();
-var position = new Vec3();
-var invParentWtm = new Mat4();
-var rotation = new Quat();
-var invParentRot = new Quat();
-var matrix = new Mat4();
-var target = new Vec3();
-var up = new Vec3();
+const scaleCompensatePosTransform = new Mat4();
+const scaleCompensatePos = new Vec3();
+const scaleCompensateRot = new Quat();
+const scaleCompensateRot2 = new Quat();
+const scaleCompensateScale = new Vec3();
+const scaleCompensateScaleForParent = new Vec3();
+const tmpMat4 = new Mat4();
+const tmpQuat = new Quat();
+const position = new Vec3();
+const invParentWtm = new Mat4();
+const rotation = new Quat();
+const invParentRot = new Quat();
+const matrix = new Mat4();
+const target = new Vec3();
+const up = new Vec3();
```

Comment:
Shouldn't these technically be `let` as they will change value?

Reply:
it's not c++ or c# type of const.
In JS const is if you don't assign anything else to it. So you can even have `const array = [];` and add and remove elements from it. As long as you don't do `array = null;` or similar

Reply:
Oh, I see. JS is weird 😅